rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -705,3 +722,76 @@ def parse_service_string(pkg_name, srv_name, message_string):
         pkg_name, srv_name + SERVICE_RESPONSE_MESSAGE_SUFFIX, response_message_string)
 
     return ServiceSpecification(pkg_name, srv_name, request_message, response_message)
+
+def parse_action_file(pkg_name, interface_filename):
+    basename = os.path.basename(interface_filename)
+    action_name = os.path.splitext(basename)[0]
+    with open(interface_filename, 'r') as h:
+        return parse_action_string(pkg_name, action_name, h.read())
+
+
+def parse_action_string(pkg_name, action_name, action_string):
+    action_blocks = action_string.split(ACTION_REQUEST_RESPONSE_SEPARATOR)
+    if len(action_blocks) != 3:
+        raise InvalidSpecification(
+            "Number of '%s' separators nonconformant with action definition" %
+            ACTION_REQUEST_RESPONSE_SEPARATOR)
+
+    goal_service_string, result_service_string, feedback_message_string = action_blocks
+
+    services = []
+    #----------------------------------------------------------------------------------------------
+    ## Send goal
+    implicit_input = ["uint8[16] uuid"]
+    request_message_string = goal_service_string + '\n' + '\n'.join(implicit_input)
+    request_message = parse_message_string(
+        pkg_name,
+        action_name + ACTION_GOAL_MESSAGE_SUFFIX + SERVICE_REQUEST_MESSAGE_SUFFIX,
+        request_message_string)
+
+    implicit_output = ["bool accepted", "builtin_interfaces/Time stamp"]
+    response_message_string = '\n'.join(implicit_output)
+    response_message = parse_message_string(
+        pkg_name,
+        action_name + ACTION_GOAL_MESSAGE_SUFFIX + SERVICE_RESPONSE_MESSAGE_SUFFIX,
+        response_message_string)
+
+    services.append(ServiceSpecification(
+        pkg_name,
+        action_name + ACTION_GOAL_MESSAGE_SUFFIX,
+        request_message,
+        response_message))
+    #----------------------------------------------------------------------------------------------
+
+    #----------------------------------------------------------------------------------------------
+    ## Get result
+    implicit_input = ["uint8[16] uuid"]
+    request_message_string = '\n'.join(implicit_input)
+    request_message = parse_message_string(
+        pkg_name,
+        action_name + ACTION_RESULT_MESSAGE_SUFFIX + SERVICE_REQUEST_MESSAGE_SUFFIX,
+        request_message_string)
+
+    implicit_output = ["int8 status"]
```

Comment:
What's the status field for?

Reply:
It's the goal status as defined in the [proposal](https://github.com/ros2/design/blob/actions_proposal/articles/actions.md#get-result-service).

Reply:
Whoops, I completely forgot about that

Reply:
@jacobperron same question about `rcl_actions` using the sequence number vs goal id. If the latter then using [action_msgs/GoalStatus](https://github.com/ros2/rcl_interfaces/blob/master/action_msgs/msg/GoalStatus.msg) here would get that.

Reply:
Same as above.